mws/response.py

```diff
@@ -0,0 +1,201 @@
+
+    @property
+    def headers(self):
+        """Shortcut to ``.original.headers``."""
+        return self.original.headers
+
+    @property
+    def encoding(self):
+        """Shortcut to ``.original.encoding``.
+        Can also be used as a setter, changing the encoding of the response.
+        This then changes how content is decoded when using :py:meth:`.text <.text>`.
+        """
+        return self.original.encoding
+
+    @encoding.setter
+    def encoding(self, val):
+        self.original.encoding = val
+
+    @property
+    def reason(self):
+        """Shortcut to ``.original.reason``."""
+        return self.original.reason
+
+    @property
+    def cookies(self):
+        """Shortcut to ``.original.cookies``."""
+        return self.original.cookies
+
+    @property
+    def elapsed(self):
+        """Shortcut to ``.original.elapsed``."""
+        return self.original.elapsed
+
+    @property
+    def request(self):
+        """Shortcut to ``.original.request``."""
+        return self.original.request
+
+
+class MWSResponse(ResponseWrapperBase):
+    """Wraps a ``requests.Response`` object and extracts some known data.
+
+    Particularly for XML responses, parsed contents can be found in the ``.parsed``
+    property as a ``DotDict`` instance.
+
+    Find metadata in ``.metadata``, mainly for accessing ``.metadata.RequestId``;
+    or simply use the ``.request_id`` shortcut attr.
+
+    :param request.Response response: Response object returned by a request sent
+     to MWS.
+    :param str result_key: Key to use as the root for ``.parsed``.
+     Typically a tag in the root of the response's XML document whose name ends
+     in ``Result``. Defaults to ``None``, in which case the full document is
+     presented when using ``.parsed``.
+    :param bool force_cdata: Passed to ``xmltodict.parse()`` when parsing
+     the response's XML document. Defaults to ``False``.
+    """
+
+    __attrs__ = [
+        "original",
+        "content",
+        "text",
+        "status_code",
+        "headers",
+        "encoding",
+        "reason",
+        "cookies",
+        "elapsed",
+        "request",
+        "parse_response",
+        "parsed",
+        "metadata",
+        "request_id",
+    ]
+
+    def __init__(self, response, result_key=None, encoding=None, force_cdata=False):
+        super().__init__(response)
+        self.timestamp = None
+        self._result_key = result_key
+
+        if not self.encoding:
+            # If the response did not specify its encoding,
+            # we use either A) an encoding specified by the user,
+            # or B) the ``apparent_encoding`` of the ``requests.Response`` object,
+            # which uses ``chardet`` to guess the encoding of the response content.
+            # Either way, we need an encoding saved in order to parse the content
+            # from XML into DotDicts.
+            self.encoding = encoding or response.apparent_encoding
```

Comment:
There is quite a discussion around encoding - is this the sensible option still ?

Reply:
This was an adjustment re: #209 discussion. In the end I decided to go with a hierarchy of:

1. Use the encoding that the response provides in its headers (if `requests` gets a header with a content encoding specified, it saves that and will decode `.text` using it). *If not provided*:
2. Use an encoding explicitly requested by the user. During `make_request`, the value of `MWS.force_response_encoding` is passed here, which defaults to `None`. If the user specified, for example, `products_api.force_response_encoding = 'latin-1'`, then that encoding will be used to decode the response first. *Or, finally*:
3. Explicitly call on charset detection using `response.apparent_encoding`, to guess the encoding of the response.